internal/indexmigration/migration.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package indexmigration
+
+import (
+	"os"
+
+	"k8s.io/klog"
+
+	"sigs.k8s.io/krew/internal/environment"
+	"sigs.k8s.io/krew/internal/gitutil"
+	"sigs.k8s.io/krew/pkg/constants"
+)
+
+// Done checks if the krew installation requires a migration to support multiple indexes.
+// A migration is necessary when the index directory doesn't contain a "default" directory.
+func Done(paths environment.Paths) (bool, error) {
+	f, err := os.Stat(paths.DefaultIndexPath())
+	if err == nil {
+		return f.IsDir(), nil
+	} else if os.IsNotExist(err) {
+		return false, nil
+	}
+	return false, err
+}
+
+// Migrate removes the index directory and then clones krew-index to the new default index path.
+func Migrate(paths environment.Paths) error {
+	isMigrated, err := Done(paths)
+	if err != nil {
+		return err
+	}
+	if isMigrated {
+		klog.Infoln("Already migrated")
```

Comment:
please use leveled logs (e.g. klog.v(2) ) for stuff users won't see.

also "already migrated" is not a good log line by itself.

Reply:
Sounds good. I've seen that throughout the code base, is there some guide for what levels should be used in certain scenarios? I briefly looked through the klog repo but didn't see anything immediately.

Reply:
Non-leveled logs: always visible to users
Leveled logs: never visible to users, unless `-v LEVEL` is specified.

We frequently use `-v 2` or `-v 10` etc for debugging.
So use judgement from the rest of the repo. :) If it's too off, we'll comment.

Reply:
Actually wouldn't we want to display to the user that the migration has already been performed?

Reply:
🤔 you’re right but this actually sounds like an error.
But both are fine since this is just a one off command.

Reply:
Ok I've already changed it to use leveled logging, so I'll just leave it that way if you think either way is cool.